### DISCUSSION TITLE
S3 Object Lambda Support

### DISCUSSION THREAD

**Discussion by drauziooppenheimer** [aws/aws-lambda-rust-runtime#479] [Q&A] [answered]
Hi, is there any way to implement S3 Object Lambdas with the current runtime and, if so, is there any documentation/example of it? 
Thanks a lot.

**Comment by calavera** [comment, marked as answer]
Yes, that's possible. You'll need to write the struct to represent the S3ObjectEvent because it doesn't look like it's defined in the `aws_lambda_events` crate. 

Looking at the examples, in [the documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/olap-writing-lambda.html), this is how the code would look like, roughly:

```rust
use lambda_runtime::{service_fn, Error, LambdaEvent};
use aws_lambda_events::s3::object_lambda::S3ObjectLambdaEvent;

pub(crate) async fn my_handler(event: LambdaEvent<S3ObjectLambdaEvent>) -> Result<(), Error> {
  let object = event.payload.;

  // Do something useful with the object here
  
  Ok(())
}
```

You can use the S3 SDK to write responses:

https://docs.rs/aws-sdk-s3/0.11.0/aws_sdk_s3/client/struct.Client.html#method.write_get_object_response
https://github.com/awslabs/aws-sdk-rust/blob/cc2359ad6e101031640dafb66307aeeb3e218e84/sdk/s3/src/operation.rs#L3452

Let me know if that helps.

**Reply by calavera** [reply]
I went ahead an implemented the event structure for you here: https://github.com/LegNeato/aws-lambda-events/pull/92

**Reply by drauziooppenheimer** [reply]
That's awesome, thanks a lot!
